### PR TITLE
Setting the cluster's name

### PR DESCRIPTION
**Pull request by raaizik**
## Changes
[RHSTOR-5500](https://issues.redhat.com/browse/RHSTOR-5500)
Setting the cluster's name that was exposed to the consumer's status in [#2534](https://github.com/red-hat-storage/ocs-operator/pull/2534).

## TODOs (later PRs)

- Refactor the `set`ters such that the code that fills in the fields is extracted into its own function
- Make the RBAC more restrictive

### CODE REVIEW

File: service/status-report/main.go [hunk 1]
```diff
@@ -132,6 +137,20 @@ func main() {
...
+	if err = cl.Get(ctx, client.ObjectKeyFromObject(clusterConfig), clusterConfig); err != nil {
+		klog.Exitf("Failed to get clusterConfig %q/%q: %v", clusterConfig.Namespace, clusterConfig.Name, err)
+	}
+	clusterMetadata := clusterConfig.Data["metadata"]
+	var clusterMetadataRes map[string]string
+	err = json.Unmarshal([]byte(clusterMetadata), &clusterMetadataRes)
+	if err != nil {
+		klog.Exitf("Fatal error, %v", err)
+	}
+	clusterName := clusterMetadataRes["name"]
```
Comment (leelavg) [inline service/status-report/main.go:152]: I don't think this is correct, as per the sample config you posted earlier there's yaml value for `install-config` key stored in the configmap and below might work
``` go
        installConfig := clusterConfig.Data["install-config"]
	metadata := struct {
		Metadata struct {
			Name string `yaml:"name"`
		} `yaml:"metadata"`
	}{}
	yaml.Unmarshal([]byte(installConfig), &metadata)
        fmt.Println(metadata.Name) // check for emptiness as well if required
```

File: service/status-report/main.go [hunk 2]
```diff
@@ -38,7 +41,9 @@ import (
 )
 
 const (
-	csvPrefix = "ocs-client-operator"
+	csvPrefix              = "ocs-client-operator"
+	clusterConfigNamespace = "kube-system"
```
[the remark this answers is not in the document]
Comment (raaizik) [reply service/status-report/main.go:45]: What do you mean by that?

[the remark this answers is not in the document]
Comment (leelavg) [reply service/status-report/main.go:44]: You are trying to read a configmap in some specific namespace and I'm just asking whether the serviceaccount (sa) that you use already has perms for that or not.

[the remark this answers is not in the document]
Comment (raaizik) [reply service/status-report/main.go:44]: I'm not sure how to setup an RBAC in order for it to be able to read this CM.

[the remark this answers is not in the document]
Comment (leelavg) [reply service/status-report/main.go:44]: Ack, I just checked, this sa already has required perms https://github.com/red-hat-storage/ocs-client-operator/blob/main/config%2Frbac%2Fstatus-reporter-clusterrole.yaml.

[the remark this answers is not in the document]
Comment (leelavg) [reply service/status-report/main.go:44]: blank is for core apis. You can check all resources that belong to core as `k api-resources --api-group=""`

[the remark this answers is not in the document]
Comment (raaizik) [reply service/status-report/main.go:44]: Right, thought it's for native resources. Now how do I know that this covers resources that reside in a system namespace?

[the remark this answers is not in the document]
Comment (leelavg) [reply service/status-report/main.go:44]: Above is a cluster role and so you can read configmap in all namespaces.

File: service/status-report/main.go [hunk 3]
```diff
@@ -18,6 +18,8 @@ package main
 
 import (
 	"context"
+	v1 "k8s.io/api/core/v1"
```
Comment (leelavg) [inline service/status-report/main.go:21]: as mentioned in previous review.
```suggestion
	corev1 "k8s.io/api/core/v1"
```

File: service/status-report/main.go [hunk 4]
```diff
@@ -132,6 +136,26 @@ func main() {
...
+	clusterMetadataRes := struct {
+		Metadata struct {
+			Name string `yaml:"name"`
+		} `yaml:"metadata"`
+	}{}
+	err = yaml.Unmarshal([]byte(clusterMetadata), &clusterMetadataRes)
+	if err != nil {
+		klog.Exitf("Fatal error, %v", err)
+	}
+	clusterName := clusterMetadataRes.Metadata.Name
```
Comment (leelavg) [inline service/status-report/main.go:157]: can we check this value isn't empty?

**Review by leelavg — COMMENTED** [review]
pls add minimal testing results. thanks.

File: service/status-report/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nb-ohad) [reply service/status-report/main.go:44]: This is a very very permissive RBAC, can we please scope it to the specific configmap in question?

[the remark this answers is not in the document]
Comment (raaizik) [reply service/status-report/main.go:44]: I'm not familiar with another way to fetch a CM that resides in that ns, @nb-ohad.

**Comment by raaizik** [comment]
> @raaizik It seems that with this change we are already trying to acquire multiple information bits before sending the response. Can we have a small refactor to separate each of these into its own function instead of just having a banch of code at the start of main? I am only referring to the section of code that try to get the information to send on the heartbeat.
> 

Is there a need to do further refactoring such that the arguments list that's being sent to those `Set`ters would comprise of only two? Meaning send in only one argument that refers to the object's information (e.g., `setClusterInformation(status, clusterInfo)`?

**Pull request merged**